linq4j/src/main/java/org/apache/calcite/linq4j/tree/ExpressionWriter.java

```diff
@@ -32,15 +32,25 @@ class ExpressionWriter {
 
   private final Spacer spacer = new Spacer(0);
   private final StringBuilder buf = new StringBuilder();
-  private boolean indentPending;
   private final boolean generics;
+  private final int methodSplittingThreshold;
+  private boolean indentPending;
 
   ExpressionWriter() {
-    this(true);
+    this(true, 0);
   }
 
-  ExpressionWriter(boolean generics) {
+  ExpressionWriter(boolean generics, int methodSplittingThreshold) {
```

Comment:
Need javadoc for this constructor.

You don't say the units of the threshold (bytes, lines), that it applies to source code length, what values disable splitting, and whether it's `>` or `>=`.

Reply:
Expanded comments and renamed this parameter.